src/EditorFeatures/Core/ValueTracking/ValueTrackingService.cs

```diff
@@ -120,54 +115,75 @@ or ILocalSymbol
                 case IFieldSymbol:
                     {
                         // The "output" is a variable assignment, track places where it gets assigned
-                        await TrackVariableSymbolAsync(previousTrackedItem.Symbol, previousTrackedItem.Document, progressCollector, cancellationToken).ConfigureAwait(false);
+                        await TrackVariableSymbolAsync(previousTrackedItem.Symbol, operationCollector, cancellationToken).ConfigureAwait(false);
                     }
                     break;
 
                 case IParameterSymbol parameterSymbol:
                     {
-                        // The "output" is method calls, so track where this method is invoked for the parameter as 
-                        // well as assignments inside the method. Both contribute to the final values
-                        await TrackVariableSymbolAsync(previousTrackedItem.Symbol, previousTrackedItem.Document, progressCollector, cancellationToken).ConfigureAwait(false);
-                        await TrackParameterSymbolAsync(parameterSymbol, previousTrackedItem.Document, progressCollector, cancellationToken).ConfigureAwait(false);
+                        var previousSymbol = previousTrackedItem.Parent?.Symbol;
+
+                        // If the current parameter is a parameter symbol for the previous tracked method it should be treated differently.
+                        // For example: 
+                        // string PrependString(string pre, string s) => pre + s;
+                        //        ^--- previously tracked          ^---- current parameter being tracked
+                        //
+                        // In this case, s is being tracked because it contributed to the return of the method. We only
+                        // want to track assignments to s that could impact the return rather than tracking the same method
+                        // twice.
+                        var isParameterForPreviousTrackedMethod = previousSymbol == parameterSymbol.ContainingSymbol;
```

Comment:
1. use .Equals.  == is not guaranteed to work in the compiler.
2. this checks for a direct 'previous'.  is there any concern about transitive previous? (i honestly dont' know).

Reply:
Good catch. Do we not have the "Compare symbols correctly" analyzer enabled? 🤔

Right now there is no known need for transitive checks, but since this is a tree would be fairly straight forward to check all ancestors